src/apps/miscapps/h3ToGeoHier.c

```diff
@@ -95,55 +97,92 @@ void recursiveH3IndexToGeo(H3Index h, int res, int isKmlOut) {
     }
 }
 
-int main(int argc, char* argv[]) {
-    // check command line args
-    if (argc < 2 || argc > 5) {
-        fprintf(stderr, "usage: %s H3Index [resolution outputMode]\n", argv[0]);
-        exit(1);
+int main(int argc, char *argv[]) {
+    int res = 0;
+    H3Index parentIndex = 0;
+    char userKmlName[BUFF_SIZE] = {0};
+    char userKmlDesc[BUFF_SIZE] = {0};
+
+    Arg helpArg = {.names = {"-h", "--help"},
+                   .helpText = "Show this help message."};
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText =
+                      "Resolution, if less than the resolution of the parent "
+                      "only the parent is printed. Default 0."};
+    Arg parentArg = {
+        .names = {"-p", "--parent"},
+        .scanFormat = "%" PRIx64,
+        .valueName = "parent",
+        .value = &parentIndex,
+        .required = true,
+        .helpText = "Print cell centers descendent from this index."};
+    Arg kmlArg = {.names = {"-k", "--kml"},
```

Comment:
Not required, but would it be worthwhile having common args defined in a util file and reused here? The KML args, the help arg, etc are pretty standard, and might be annoying to maintain across multiple files.